planet/src/main/java/com/google/planet/servlets/EventServlet.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.planet.servlets;
+import com.google.gson.Gson;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.planet.data.Event;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import java.util.Arrays;
+import java.util.ArrayList;
+import java.util.List;
+import java.lang.*; 
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that handles the game two truths and one lie */
+
+@WebServlet("/update-event")
+public class EventServlet extends HttpServlet {
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        // Wait 50 ms for datastore to update changes
+        try{
+            Thread.sleep(50);
+        }catch(InterruptedException ex){
+            Thread.currentThread().interrupt();
+        }
+
+        Query query = new Query("Event");
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        List<Event> events = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            long id = entity.getKey().getId();
+            String name = (String) entity.getProperty("name");
+            String address = (String) entity.getProperty("address");
+            double duration = (double) entity.getProperty("duration");
+            String preferredTime = (String) entity.getProperty("preferredTime");
```

Comment:
Looks like you can store date and time directly as a valuetype: https://cloud.google.com/appengine/docs/standard/java/datastore/entities#Properties_and_value_types

Reply:
Thank you! I tried adding the sleep function to doPost but unfortunately it wasn't working when I tested it (the datastore doesn't update in time), will try it again. We are also switching to firebase instead of datastore later.